applications/vessel-history/src/features/map/Map.tsx

```diff
@@ -0,0 +1,85 @@
+import { ReactElement, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { InteractiveMap } from '@globalfishingwatch/react-map-gl'
+import { useLayerComposer } from '@globalfishingwatch/react-hooks'
+import {
+  getLayerComposerLayers,
+  // getMapboxPaintIcon,
+  // selectDirectionPointsLayers,
+  // selectLegendLabels,
+} from './map.selectors'
+import {
+  useGeneratorsConnect,
+  // useMapBounds,
+  // useMapClick,
+  // useMapMove,
+  // useViewport,
+} from './map.hooks'
+import { useMapboxRef, useMapboxRefCallback } from './map.context'
+import styles from './Map.module.css'
+
+const Map = (): ReactElement => {
+  const mapRef = useMapboxRef()
+
+  const onRefReady = useMapboxRefCallback()
+  // const { viewport, onViewportChange } = useViewport()
+
+  const generatorConfigs = useSelector(getLayerComposerLayers)
+  const { globalConfig } = useGeneratorsConnect()
+  const [viewport, setViewport] = useState({
+    latitude: 37.7577,
+    longitude: -122.4376,
+    zoom: 8,
+  })
+
+  const { style } = useLayerComposer(generatorConfigs, globalConfig)
+
+  // return (
+  //   <ReactMapGL
+  //     {...viewport}
+  //     width="100%"
+  //     height="100%"
+  //     onViewportChange={(viewport) => setViewport(viewport)}
+  //   />
+  // )
+  return (
+    <div className={styles.container}>
+      {style && (
+        <InteractiveMap
+          disableTokenWarning={true}
+          ref={mapRef}
+          width="100%"
+          height="100%"
+          latitude={viewport.latitude}
+          longitude={viewport.longitude}
+          zoom={viewport.zoom}
+          // onLoad={onLoadCallback}
+          // onViewportChange={onViewportChange}
+          // mapStyle={styleWithArrows}
+          // onClick={handleMapClick}
+          // onHover={handleMapHover}
+          // onMouseMove={onMapMove}
+          mapOptions={{
```

Comment:
Just to not forget, it is better to declare the config as a const outside of the render to avoid re-rendering the entire map on every change as it is a new object instance